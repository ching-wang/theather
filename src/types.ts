import React from "react";

export type Show = {
    id: number,
    url: string,
    name: string;
    genres: string[],
    language: string,
    runtime: number,
    rating: object,
}

export type ReactChangeEvent = React.ChangeEvent<HTMLInputElement>

export type InputHandler = (event: ReactChangeEvent) => void

export type SearchResult = {
  score: number;
  show: Show;
};
