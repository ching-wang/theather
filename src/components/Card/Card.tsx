import React from "react";
import { Link } from "react-router-dom";
import { Show } from "../../types"
import { CardWrapper, Thumbnail, Title} from "./card.styles";


export const Card = ({ show } : {show: Show} ) => {
    return (
        <CardWrapper>
            <Link key={show.id} to={`/show/${show.id}`}>
                <Thumbnail src={show?.image?.medium} alt="thumbnail" />
                <Title>
                    {show.name}
                </Title>
            </Link>
        </CardWrapper>
    )
}

export default Card;