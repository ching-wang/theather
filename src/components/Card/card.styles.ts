import styled from "styled-components";
import {theme} from "../../styles/GlobalStyles";

export const CardWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  box-shadow: ${theme.shadow};
  width: 250px;
  margin: 10px;
  padding: 20px;
`;

export const Thumbnail = styled.img`
  width: 200px;
  padding: 5px;
`;

export const Title = styled.p`
  font-size: 1.5rem;
`;
