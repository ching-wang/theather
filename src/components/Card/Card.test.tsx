import React from "react";
import Card from "./Card";
import { Show } from "../../types";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe(Card.name, () => {
  it("renders", () => {
    const mockShow: Show = {
      id: 1,
      url: "http://hellothere.com",
      name: "foobar",
      genres: [],
      language: "",
      runtime: 123,
      rating: {
        average: "4",
      },
      number: 1,
      status: "good",
      summary: "foobar",
      image: {
        medium: "foobar.com/img.jpg",
        original: "foobar.com/img.jpg",
      },
    };
    const tree = renderer
      .create(
        <BrowserRouter>
          <Card show={mockShow} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
