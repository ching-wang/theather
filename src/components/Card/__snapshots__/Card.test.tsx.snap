// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<div
  className="sc-bdfBQB eQDmeF"
>
  <a
    href="/show/1"
    onClick={[Function]}
  >
    <img
      alt="thumbnail"
      className="sc-gsTEea CMgeu"
      src="foobar.com/img.jpg"
    />
    <p
      className="sc-dlfnuX oYBdS"
    >
      foobar
    </p>
  </a>
</div>
`;
